src/services/query-options.js

```diff
@@ -0,0 +1,256 @@
+import { logger, Schemas } from 'forest-express';
+import _ from 'lodash';
+import Operators from '../utils/operators';
+import QueryUtils from '../utils/query';
+import CompositeKeysManager from './composite-keys-manager';
+import { ErrorHTTP422 } from './errors';
+import FiltersParser from './filters-parser';
+import LiveQueryChecker from './live-query-checker';
+import QueryBuilder from './query-builder';
+import SearchBuilder from './search-builder';
+
+/**
+ * Sequelize query options generator which is configured using forest admin concepts (filters,
+ * search, segments, ...).
+ * Those can be used for update, findAll, destroy, ...
+ */
+class QueryOptions {
+  /**
+   * Query options which can be used with sequelize.
+   * i.e: Books.findAll(queryOptions.sequelizeOptions);
+   */
+  get sequelizeOptions() {
+    // Used for failed search, and when retricting to empty lists of ids.
+    // Saves us from having to deal with thoses cases in all charts, ressources-getter, ...
+    if (this._returnZeroRecords) {
+      const Sequelize = this._model.sequelize.constructor;
+      return { where: Sequelize.literal('(0=1)') };
+    }
```

Comment:
On some collection, search is not supported, because none of the fields are searchable.
This piece of code is there to ensure that no results are returned when that is the case.

Previouly it was implemented here: 
https://github.com/ForestAdmin/forest-express-sequelize/blob/master/src/services/resources-getter.js#L157-L166

I was copy pasted on the count() and perform() so I think it was a bugfix done at some point.
I moved it in the data layer because it's neater and allow to handle other cases.